packages/flutter_tools/test/general.shard/resident_runner_test.dart

```diff
@@ -599,6 +605,44 @@ void main() {
 
     expect(await fs.file('foo').readAsString(), testUri.toString());
   }));
+
+  test('FlutterDevice uses dartdevc configuration when targeting web', () => testbed.run(() async {
+    final MockDevice mockDevice = MockDevice();
+    final StreamController<List<int>> stdout = StreamController<List<int>>();
+    when(mockDevice.targetPlatform).thenAnswer((Invocation invocation) async {
+      return TargetPlatform.web_javascript;
+    });
+    when(processManager.start(any)).thenAnswer((Invocation invocation) async {
+      final List<String> arguments = invocation.positionalArguments.first;
+      // Contains configuration for dartdevc target.
+      expect(arguments,
+        contains(artifacts.getArtifactPath(Artifact.flutterWebSdk, mode: BuildMode.debug) + '/'));
+      expect(arguments,
+        contains(artifacts.getArtifactPath(Artifact.webPlatformKernelDill, mode: BuildMode.debug)));
+      expect(arguments,
+        contains('--target=dartdevc'));
+      return FakeProcess(
+        stdout: stdout.stream,
+      );
+    });
+
+    final FlutterDevice flutterDevice = await FlutterDevice.create(
+      mockDevice,
+      buildMode: BuildMode.debug,
+      flutterProject: FlutterProject.current(),
+      target: null,
+      trackWidgetCreation: true,
+    );
+    final Future<void> onDone = flutterDevice.generator.recompile('main.dart', <Uri>[], outputPath: 'app.dill');
+    stdout.add(utf8.encode('result 578145e3-fe1c-4094-9dda-c208d5a60168\n'));
+    stdout.add(utf8.encode('578145e3-fe1c-4094-9dda-c208d5a60168\n'));
+    stdout.add(utf8.encode('+file:///foo/flutter/packages/flutter/lib/src/physics/utils.dart\n'));
+    stdout.add(utf8.encode('578145e3-fe1c-4094-9dda-c208d5a60168 app.dill 0\n'));
+    await onDone;
```

Comment:
Is there any state on `flutterDevice` or `flutterDevice.generator` that can be checked for correctness here?

Reply:
ResidentCompiler doesn't expose any of its configuration - it could, that might be an easier test to confirm the artifact locations are correct.

Reply:
Updated ResidentCompiler to be an interface, with a visibleForTesting implementation that I can look at the getters of